src/Util/LocaleCodeProvider.php

```diff
@@ -68,15 +68,28 @@ public function getFormattedLocaleCode(string $localeCode): string
         $canonicalizedCode = (string) \Locale::canonicalize($localeCode);
 
         if (\mb_strlen($canonicalizedCode) !== self::SUPPORTED_LOCALE_CODE_LENGTH) {
+            $matched = $this->findMatchingSupportedLocale($canonicalizedCode);
```

Comment:
We need to find a suitable locale for any code, not just those that are too long